### DISCUSSION TITLE
AppIntro starts everytime App launches

### DISCUSSION THREAD

**Discussion by hiriworst** [AppIntro/AppIntro#1044] [Q&A] [answered]
I need baby steps on how i can deal with AppIntro starting everytime i launch App. I declareed it in App Activty as last but how and where to implement the SharedPreference where i get lost.. I have no idea how to.. Please help!

**Comment by cortinico** [comment, marked as answer]
There are a number of different ways to do so.

You can use SharedPreferences to store if the user saw the AppIntro or not (say `user_saw_intro`)
You will then have to make sure the first activity is not AppIntro. You can have like a splash screen that loads AppIntro if `user_saw_intro` is false, and skips to the home page otherwise

**Reply by hiriworst** [reply]
I have tried using some sharedpreference code but am caught in an endless loop.. i think I'll paste the code here for debug
